src/test_typing_extensions.py

```diff
@@ -5779,6 +5779,42 @@ class D(Protocol[T1, T2, Unpack[Ts]]): pass
                     with self.assertRaises(TypeError):
                         klass[int]
 
+    def test_substitution(self):
+        Ts = TypeVarTuple("Ts")
+        unpacked_str = Unpack[Ts][str]
+        with self.subTest("Check substitution result"):
+            self.assertIs(unpacked_str, str)
+
+    @skipUnless(TYPING_3_11_0, "Needs Issue #103 for <3.11")
+    def test_nested_unpack(self):
+        T = TypeVar('T')
+        Ts = TypeVarTuple("Ts")
+        Variadic = Tuple[int, Unpack[Ts]]
+        # Tuple[int, int, Tuple[str, int]]
+        direct_subscription = Variadic[int, Tuple[str, int]]
+        # Tuple[int, int, Tuple[*Ts, int]]
+        TupleAliasTs = Variadic[int, Tuple[Unpack[Ts], int]]
+
+        # if this fails all below are likely to fail too
+        # Tuple[int, int, Tuple[str, int]]
+        recursive_unpack = TupleAliasTs[str]
+        self.assertEqual(direct_subscription, recursive_unpack)
+        self.assertEqual(get_args(recursive_unpack), (int, int, Tuple[str, int]))
+
+        TupleAliasTsT = Variadic[Tuple[Unpack[Ts], T]]
+        # Equivalent Forms
+        with self.subTest("Equivalence of variadic arguments"):
```

Comment:
Only use subTest in a loop. This one should probably just be a separate test method.